apps/evm/lib/evm/builtin.ex

```diff
@@ -219,24 +220,107 @@ defmodule EVM.Builtin do
       gas < @g_ec_mult ->
         {0, %EVM.SubState{}, exec_env, :failed}
 
-      x > IntegerModP.default_modulus() || y > IntegerModP.default_modulus() ->
+      x > FQ.default_modulus() || y > FQ.default_modulus() ->
         {0, %EVM.SubState{}, exec_env, :failed}
 
       true ->
         calculate_ec_mult({x, y}, scalar, gas, exec_env)
     end
   end
 
+  @spec ec_pairing(EVM.Gas.t(), EVM.ExecEnv.t()) ::
+          {EVM.Gas.t(), EVM.SubState.t(), EVM.ExecEnv.t(), EVM.VM.output()}
+  def ec_pairing(gas, exec_env) do
+    data = exec_env.data
+    pair_size = byte_size(exec_env.data)
+
+    gas_cost = @g_ec_pairing_point * div(pair_size, 192) + @g_ec_pairing
```

Comment:
Where does the `192` come from? Would it make sense to extract that to a named variable to give it some more context?